src/type/scalars.js

```diff
@@ -66,13 +66,19 @@ export const GraphQLInt = new GraphQLScalarType({
   serialize: serializeInt,
   parseValue: coerceInt,
   parseLiteral(ast) {
-    if (ast.kind === Kind.INT) {
-      const num = parseInt(ast.value, 10);
-      if (num <= MAX_INT && num >= MIN_INT) {
-        return num;
-      }
+    const value = ast.value || '';
```

Comment:
@danielrearden It means you can get `Int cannot represent non-integer value: ""` for some node types (e.g. null nodes) which would be very confusing.
It's better to write it like this:
```js
  parseLiteral(ast) {
    if (ast.kind !== Kind.INT) {
      throw new TypeError(
        'Int cannot represent non-integer value' +
          (ast.value ? ': ' + inspect(ast.value) : '.'),
      );
    }
    const num = parseInt(ast.value, 10);
    // ...
  },
```
Can you please change it here and in other places?
Also please write add tests for such cases (AST nodes without values) so we would maintain same test coverage:
https://codecov.io/gh/graphql/graphql-js/commit/92d58e750257cabb84361fb64e9b6306ba91e57c#D2-69

Reply:
@IvanGoncharov Sorry for the late response. Finally had a bit of time to circle back around to this PR. I'll make the necessary changes, but after taking another pass at it, I'm wondering if we could be better around reducing the code duplication between the two `parse` methods.

I'm thinking we could convert the `ValueNode` that's passed into `parseLiteral` into a regular JS value and then pass that to `parseValue`. That way there's a single function responsible for input validation. There's already a `valueFromASTUntyped` utility function available, so it could be as simple as:

```
parseLiteral: (ast, variables) => coerceString(valueFromASTUntyped(ast, variables))
```

The main gotcha I can think of is that `1.0` would get coerced into `1` so we'd have to still explicitly throw on float values for `GraphQLInt` and `GraphQLID`.

Reply:
@danielrearden Let's keep this PR around replacing `undefined` with `throw`.
On a topic why we need both `parseLiteral` and `parseValue` as separate function see this comment: https://github.com/graphql/graphql-js/issues/500#issuecomment-478193989